### PR TITLE
Export selected columns

### PR DESCRIPTION
**Pull request by dvoet**
https://broadinstitute.atlassian.net/browse/GAWB-1232
Added an optional parameter to tsv download end point to specify attribute names, in order, to be in the resulting tsv. Note that this does not trickle down to rawls so the full data set still goes over the wire between orch and rawls.... meh. The first commit deals with just about all of this.

Also refactored the areas I touched to match new orch structure. The refactor also fixed https://broadinstitute.atlassian.net/browse/GAWB-1305 because the new code does not custom parse json responses and I think the regression came with the attribute list format refactor.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala [hunk 1]
```diff
@@ -11,102 +10,98 @@ import spray.json.JsValue
...
+
+    val headers: IndexedSeq[String] = entityHeader +: requestedHeadersSansId.getOrElse(defaultHeaders(entityType, filteredEntities, headerRenamingMap))
+    val rows: IndexedSeq[IndexedSeq[String]] = filteredEntities.filter { _.entityType == entityType }
       .map { entity =>
         makeRow(entity, headers, headerRenamingMap)
       }.toIndexedSeq
     exportToString(headers, rows)
   }
 
+  def defaultHeaders(entityType: String, filteredEntities: Seq[RawlsEntity], headerRenamingMap: Map[String, String]) = {
```
Comment (breilly2) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala:62]: Nice extracted function and simplification. I'll have to remember to keep `collect` in mind.

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala [hunk 2]
```diff
@@ -11,102 +10,98 @@ import spray.json.JsValue
...
+    * @return new list of RawlsEntity
     */
-  private def filterAttributeFromEntities(entities: List[EntityWithType], attributeName: String): List[EntityWithType] = {
+  private def filterAttributeFromEntities(entities: Seq[RawlsEntity], attributeName: String): Seq[RawlsEntity] = {
     entities map {
       entity =>
-        val attributes = entity.attributes.getOrElse(Map.empty) filterNot {
-          attribute => attribute._1.equals(attributeName)
+        val attributes = entity.attributes filterNot {
+          case (thisAttributeName, _) => thisAttributeName == AttributeName.withDefaultNS(attributeName)
```
Comment (breilly2) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala:86]: Another nice trick I'll have to keep in mind.

**Comment by breilly2** [comment]
👍 Good test coverage added for GAWB-1232.
I'm not certain that I see how GAWB-1305 was fixed. Does it need separate testing or is the problem surfaced in the existing tests now that they're using the Attribute\* case classes/objects (or maybe something completely different)?

**Comment by breilly2** [comment]
I also appreciate the refactoring. Took a bit of comparison to understand what changed, but I think the pattern that you're pushing forward makes sense, and it's what I would expect after working in Rawls.

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala [hunk 3]
```diff
@@ -11,102 +10,98 @@ import spray.json.JsValue
...
-    val rows: List[IndexedSeq[String]] = entities.filter { _.entityType == entityType }.flatMap {
+    val rows: Seq[IndexedSeq[String]] = entities.filter { _.entityType == entityType }.flatMap {
       entity =>
-        entity.attributes.getOrElse(Map.empty).filter {
+        entity.attributes.filter {
           // To make the membership file, we need the array of elements that correspond to the set type.
           // All other top-level properties are not necessary and are only used for the data load file.
-          attribute => attribute._1.equals(collectionMemberType) }.flatMap {
-          m =>
-            Try(m._2.convertTo[List[JsValue]]) match {
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala:23]: @breilly2 GAWB-1305 was fixed here. The old code attempted to treat the attribute value as a list and if it couldn't just return IndexedSeq.empty (which is why the file was empty). The new way doesn't touch json so the problem is no longer possible.

**Pull request merged**